WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController.swift

```diff
@@ -1221,7 +1229,32 @@ extension NotificationDetailsViewController: SuggestionsTableViewDelegate {
     }
 }
 
+// MARK: - Milestone notifications
+//
+private extension NotificationDetailsViewController {
 
+    /// Determines if the notification content is a view milestone
+    var isViewMilestone: Bool {
+        FeatureFlag.milestoneNotifications.enabled && note.type == "view_milestone"
+    }
+
+    func showConfettiIfNeeded() {
+        guard FeatureFlag.milestoneNotifications.enabled,
+              isViewMilestone,
+              !confettiWasShown,
+              let view = UIApplication.shared.mainWindow,
+              let frame = navigationController?.view.frame else {
+
+            return
+        }
+
+        // removing this instance only after the animation completes, will prevent
+        // the animation to suddenly stop if users navigate away from the note
+        ConfettiView.cleanupAndAnimate(on: view, frame: frame) { $0.removeFromSuperview() }
```

Comment:
How about if the ConfettiView just removes itself from the superview in its own animation completion method? 🤔

Reply:
You mean in the convenience method? Sure!

Reply:
So I mean that we don't need to pass `$0.removeFromSuperview` here, and in `ConfettiView`'s `animationDidStop` we just add a call to `removeFromSuperview`?

Reply:
As discussed on slack, I added some clarifications and changed the implicit parameter with an explicit one, to better clarify the method.